### DISCUSSION TITLE
minimizing the need for localization files

### DISCUSSION THREAD

**Discussion by oak93** [i18next/next-i18next#1963] [Q&A]
hi everyone,

we need to support 4 languages in our application.
you can see our locales list here: 

```
locales: [
      'en-us',
      'en-es',
      'en-de',
      'en-mx',
      'en-co',
      'es-es',
      'es-de',
      'es-mx',
      'es-co',
      'de-es',
      'de-de',
      'de-mx',
      'de-co',
      'mx-es',
      'mx-de',
      'mx-mx',
      'mx-co'
    ]
```

this creates many folders inside our `public/locales` folder. 

<img width="338" alt="image" src="https://user-images.githubusercontent.com/35132536/194497173-26c675d7-adba-41d8-a6ed-576a39cfb409.png">

the language will be the same for all countries. for example, `de-es` will have the same language as `de-co`.

I have tried to use `localeSubpaths` to solve this having lots of folder issues but it did not work.
what I want to have is, I want to have only one folder for each language and serve it for all locales. 
for example serve `de` folder inside of the `public/locales` for each `de-de, de-co, de-es, de-mx`

```
// DID NOT WORK

localeSubpaths: {
      'en-us': 'en',
      'en-es': 'en',
      'en-ds': 'en',
      'en-mx': 'en',
      'en-co': 'en',
      'es-es': 'es',
      'es-de': 'es',
      'es-mx': 'es',
      'es-co': 'es',
      'de-es': 'de',
      'de-de': 'de',
      'de-mx': 'de',
      'de-co': 'de',
      'mx-es': 'mx',
      'mx-de': 'mx',
      'mx-mx': 'mx',
      'mx-co': 'mx'
    },
```

how can I achieve this? 
thanks for helping

**Comment by chrilis** [comment]
Hi @oak93, this should be possible with [Fallback locales](https://github.com/i18next/next-i18next#fallback-locales). Especially `nonExplicitSupportedLngs` might be what you are looking for.
